src/Raven.Server/Documents/Indexes/Test/BoxedItemEnumerator.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.Indexes.Test;
+
+public class BoxedItemEnumerator : IIndexedItemEnumerator
```

Comment:
What does "boxed" mean? What was the intention? Think about the reasoning why we have it and try to name it better